src/io/memory.cr

```diff
@@ -336,6 +336,24 @@ class IO::Memory < IO
     @pos = value.to_i
   end
 
+  # Sets the current bytesize to be equal to the current position
+  # effectively truncating the contents at the current position.
+  #
+  # Note this does nothing if the curront position is at the end.
+  #
+  # ```
+  # io = IO::Memory.new "hello world"
+  # io.seek(-6, Seek::Current)
+  # io.truncate
+  # io.pos = 0
+  # io.gets # => "hello"
+  # ```
+  def truncate : Nil
+    check_open
+    check_resizeable
+    @bytesize = @pos if @bytesize != @pos
```

Comment:
The `if` doesn't have any purpose...

Reply:
My perspective was a comparison is faster then a set, so don't set it if there is no difference.

Reply:
You know what, fair. A low cost applied every time to speed up bad implementations doesn't make sense.

Reply:
The comparison will bring the data into the L1d CPU cache anyway, and considering modern superscalar architectures, i'm sure they optimize a no-op memory write to L1d to have no effect on cache coherency.

I wouldn't be surprised if this is faster without the comparison.